lighter/utils/freezer.py

```diff
@@ -0,0 +1,84 @@
+from typing import List, Optional, Union
+
+from loguru import logger
+
+from lighter.utils.misc import ensure_list
+
+
+class LighterFreezer:
```

Comment:
Overall very nice. Maybe add type hints to `__call__`?

Reply:
Thanks! Yes, I forgot!